decompiler/task.py

```diff
@@ -1,113 +1,54 @@
 """Module describing tasks to be handled by the decompiler pipleline."""
 
-from typing import Dict, List, Optional
+from dataclasses import dataclass, field
+from typing import List
 
 from decompiler.structures.ast.syntaxtree import AbstractSyntaxTree
 from decompiler.structures.graphs.cfg import ControlFlowGraph
+from decompiler.structures.pseudo import Integer
 from decompiler.structures.pseudo.complextypes import ComplexTypeMap
 from decompiler.structures.pseudo.expressions import Variable
-from decompiler.structures.pseudo.typing import Integer, Type
+from decompiler.structures.pseudo.typing import Type
 from decompiler.util.options import Options
 
 
+# We set eq=False, so that tasks are only equal when they are the very same instance
+@dataclass(eq=False)
 class DecompilerTask:
     """Represents a task for the decompiler pipeline."""
 
-    def __init__(
-        self,
-        name: str,
-        cfg: ControlFlowGraph = None,
-        ast: Optional[AbstractSyntaxTree] = None,
-        options: Optional[Options] = None,
-        function_return_type: Type = Integer(32),
-        function_parameters: Optional[List[Variable]] = None,
-        complex_types: Optional[ComplexTypeMap] = None,
-    ):
-        """
-        Init a new decompiler task.
-
-        :param name -- The name of the function or task
-        :param cfg -- The control flow graph of the function
-        :param function_return_type -- The type of the return value of the decompiled function
-        :param function_parameters -- List of function parameters as Variables
-        """
-        self._name = name
-        self._cfg = cfg
-        self._ast = ast
-        self._function_return_type = function_return_type
-        self._function_parameters = function_parameters if function_parameters else []
-        self._options: Options = options if options else Options.load_default_options()
-        self._failed = False
-        self._failure_origin = None
-        self._complex_types = complex_types if complex_types else ComplexTypeMap()
-        self._code = None
-
-    @property
-    def name(self) -> str:
-        """Return the name of the task."""
-        return self._name
-
-    @property
-    def graph(self) -> ControlFlowGraph:
-        """Return a graph representing the function control flow."""
-        return self._cfg
-
-    @property
-    def syntax_tree(self) -> AbstractSyntaxTree:
-        """Return a syntax tree representing the function."""
-        return self._ast
+    name: str
+    # We allow fronted specific identifiers here. The fronted should check if it can handle a specific type of identifier
+    function_identifier: object
+    options: Options = field(default_factory=Options.load_default_options)
+    cfg: ControlFlowGraph | None = None
```

Comment:
Is this a preference which we have for the repo? I choose to use the union sytanx, because at some point I read in a [discussion](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/6) and in the [python documenation](url), that the union syntax is recommended over the typing classes like `Optional` and `Union`.
But if we prefer `Optional`, I can easily changes this.